libs/providers/go-feature-flag-web/src/lib/go-feature-flag-web-provider.ts

```diff
@@ -0,0 +1,341 @@
+  }
+
+  resolveNumberEvaluation(flagKey: string): ResolutionDetails<number> {
+    return this.evaluate(flagKey, 'number')
+  }
+
+  resolveObjectEvaluation<T extends FlagValue>(flagKey: string): ResolutionDetails<T> {
+    return this.evaluate(flagKey, 'object')
+  }
+
+  resolveStringEvaluation(flagKey: string): ResolutionDetails<string> {
+    return this.evaluate(flagKey, 'string')
+  }
+
+  resolveBooleanEvaluation(flagKey: string): ResolutionDetails<boolean> {
+    return this.evaluate(flagKey, 'boolean')
+  }
+
+  private evaluate<T extends FlagValue>(flagKey: string, type: string): ResolutionDetails<T> {
+    const resolved = this._flags[flagKey];
+    if (!resolved) {
+      throw new FlagNotFoundError(`flag key ${flagKey} not found in cache`);
+    }
+
+    if (typeof resolved.value !== type) {
+      throw new TypeMismatchError(`flag key ${flagKey} is not of type ${type}`);
+    }
+    return {
+      variant: resolved.variant,
+      value: resolved.value as T,
+      flagMetadata: resolved.flagMetadata,
+      errorCode: resolved.errorCode,
+      errorMessage: resolved.errorMessage,
+      reason: this._websocket?.readyState !== WebSocket.OPEN ? StandardResolutionReasons.CACHED : resolved.reason,
+    };
+  }
+
+  private async retryFetchAll(ctx: EvaluationContext, flagsChanged: string[] = []) {
+    let delay = this._retryInitialDelay;
+    let attempt = 0;
+    while (attempt < this._maxRetries) {
+      attempt++;
+      try {
+        await this.fetchAll(ctx, flagsChanged);
+        this._status = ProviderStatus.READY;
+        return
+      } catch (err) {
+        this._status = ProviderStatus.ERROR;
+        this.handleFetchErrors(err)
+        await new Promise((resolve) => setTimeout(resolve, delay));
+        delay *= this._retryDelayMultiplier;
+        this._logger?.info(`xxx ${delay} ms (${attempt}/${this._maxRetries}).`)
+      }
+    }
+  }
+
+  /**
+   * fetchAll is a function that is calling GO Feature Flag to bulk evaluate flags.
+   * It emits an event to notify when it is ready or on error.
+   *
+   * @param context - The static evaluation context
+   * @param flagsChanged - The list of flags update - default: []
+   * @private
+   */
+  private async fetchAll(context: EvaluationContext, flagsChanged: string[] = []) {
+    const endpointURL = new URL(this._endpoint);
+    const path = 'v1/allflags';
+    endpointURL.pathname = endpointURL.pathname.endsWith('/') ? endpointURL.pathname + path : endpointURL.pathname + '/' + path;
+
+    const request: GoFeatureFlagAllFlagRequest = {evaluationContext: transformContext(context)};
+    const headers = new Headers({
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    });
+    if(this._apiKey){
+      headers.set('Authorization', `Bearer ${this._apiKey}`);
+    }
+
+    const init: RequestInit = {
+      method: "POST",
+      headers: {
+        'Content-Type': 'application/json',
```

Comment:
You could extract the common headers to a template object, or even add them into a convenience method.